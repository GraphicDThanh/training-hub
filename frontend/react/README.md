# React Redux Training
[Plan doc](https://docs.google.com/document/d/1lYywN0RIhrwsyFZ8uPZuaN5QM78mYuXFD608vS8YEbM/edit?pli=1#)
## ES6
## React Basic
### Practice 1 - examples React
## React Advanced
### Practice 2
## Redux Basic
## Redux Advanced
## Redux Recipes
