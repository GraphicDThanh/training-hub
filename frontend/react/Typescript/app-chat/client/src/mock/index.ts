export const user = {
  id: '5ec57519-a2e2-538c-9469-8f6a5b86e671',
  name: 'Thanh',
  avatar: 'https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg'
};

export const currentUser = {
  id: '5ec57519-a2e2-538c-9469-8f6a5b86e671',
  name: 'Thanh',
  avatar: 'https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg',
  isCurrent: true
};

export const message = {
  author: {
    id: '5ec57519-a2e2-538c-9469-8f6a5b86e671',
    name: 'Thanh',
    avatar: 'https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg'
  },
  content: 'Hello',
  id: '5f8c8d9e-20e2-5be5-adfd-d107ce707209'
};

export const users = [
  {
    id: '5ec57519-a2e2-538c-9469-8f6a5b86e671',
    name: 'Thanh',
    avatar: 'https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg',
  },
  {
    id: 'b5c00d9e-c0ec-50db-a68d-9f322b3b4f5e',
    name: 'Huy',
    avatar: 'http://anhbiafb.com/uploads/images/tuyen-tap-25-avatar-doi-cuc-dang-yeu-danh-cho-cac-cap-doi-dang-yeu-nhau-1485163164-9.jpg'
  },
  {
    id: '2f397d90-8fb0-5c01-8032-03f621781f9c',
    name: 'Minh',
    avatar: 'https://tophinhanhdep.com/wp-content/uploads/2017/07/hinh-avatar-dep-de-thuong-cute-nhat-231x300.jpg'
  }
];

export const usersHaveCurrent = [
  {
    id: '5ec57519-a2e2-538c-9469-8f6a5b86e671',
    name: 'Thanh',
    avatar: 'https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg',
    isCurrent: true,
  },
  {
    id: 'b5c00d9e-c0ec-50db-a68d-9f322b3b4f5e',
    name: 'Huy',
    avatar: 'http://anhbiafb.com/uploads/images/tuyen-tap-25-avatar-doi-cuc-dang-yeu-danh-cho-cac-cap-doi-dang-yeu-nhau-1485163164-9.jpg'
  },
  {
    id: '2f397d90-8fb0-5c01-8032-03f621781f9c',
    name: 'Minh',
    avatar: 'https://tophinhanhdep.com/wp-content/uploads/2017/07/hinh-avatar-dep-de-thuong-cute-nhat-231x300.jpg'
  }
];

export const messages = [
  {
    author: {
      id: '5ec57519-a2e2-538c-9469-8f6a5b86e671',
      name: 'Thanh',
      avatar: 'https://tophinhanhdep.com/wp-content/uploads/2017/07/hinh-avatar-dep-de-thuong-cute-nhat-231x300.jpg'
    },
    content: 'Hello',
    id: '5f8c8d9e-20e2-5be5-adfd-d107ce707209'
  },
  {
    author: {
      id: 'b5c00d9e-c0ec-50db-a68d-9f322b3b4f5e',
      name: 'Huy',
      avatar: 'http://anhbiafb.com/uploads/images/tuyen-tap-25-avatar-doi-cuc-dang-yeu-danh-cho-cac-cap-doi-dang-yeu-nhau-1485163164-9.jpg'
    },
    content: 'Hi there',
    id: '33598716-296e-5224-9d49-7655f54f5ed2'
  },
];

