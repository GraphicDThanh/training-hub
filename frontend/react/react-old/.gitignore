node_modules
out
dist
.DS_Store
.git
.docpad.db
vendor/
.vagrant
bower-components/
bower_components/
npm-debug.log
.gitignore~
vagrant/Vagrantfile~
build/
