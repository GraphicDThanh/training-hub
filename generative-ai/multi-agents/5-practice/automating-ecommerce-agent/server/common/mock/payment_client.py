
import uuid
import random
from common.types.response import PaymentResponse, RefundResponse


class MockPaymentClient:
    def pay(self, payment_data) -> PaymentResponse:
        """
        Simulate processing a payment.

        Args:
            payment_data: A dictionary containing payment details.

        Returns:
            A mock response simulating a successful payment confirmation.
        """
        success_response = {
            "id": f"pay_{uuid.uuid4()}",
            "status": "succeeded",
            "amount": payment_data["amount"],
            "currency": payment_data["currency"],
            "message": "Payment processed successfully"
        }

        failure_response = {
            "id": f"pay_{uuid.uuid4()}",
            "status": "failed",
            "amount": payment_data["amount"],
            "currency": payment_data["currency"],
            "message": "Payment processing failed"
        }

        return random.choices(
            population=[success_response, failure_response],
            weights=[0.8, 0.2],
        )[0]

    def refund(self, refund_data) -> RefundResponse:
        """
        Simulate processing a refund.

        Args:
            refund_data: A dictionary containing refund details.

        Returns:
            A mock response simulating a successful refund confirmation.
        """
        success_response = {
            "id": f"refund_{uuid.uuid4()}",
            "status": "succeeded",
            "amount": refund_data["amount"],
            "currency": refund_data["currency"],
            "message": "Refund processed successfully"
        }
        failure_response = {
            "id": f"refund_{uuid.uuid4()}",
            "status": "failed",
            "amount": refund_data["amount"],
            "currency": refund_data["currency"],
            "message": "Refund processing failed"
        }
        return random.choices(
            population=[success_response, failure_response],
            weights=[0.8, 0.2],
        )[0]
