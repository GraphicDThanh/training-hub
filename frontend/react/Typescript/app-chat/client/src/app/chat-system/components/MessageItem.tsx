import * as React from 'react';
import { Message } from '../../../types/';
import { PropsSendToMessage } from './ChatSystem';
import { DEFAULT_AVATAR_URL } from '../../../constants';

interface Props extends PropsSendToMessage {
  message: Message;
}

export const MessageItem = (props: Props) => {
  if (!props.message.author.id) {
    return null;
  }
  // use destructure of ES6 to get and rename if necessary
  const {
    message: {
      author: { id, avatar, name },
      content
    },
    currentUser: { id: currentId }
  } = props;

  let classNameMessage = 'message-item';

  if (id === currentId) {
    classNameMessage += ' owner';
  }

  return (
    <div className={classNameMessage}>
      <div className="avatar-wrapper">
        <img src={avatar || DEFAULT_AVATAR_URL} alt="avatar user" className="avatar"/>
      </div>
      <div className="info">
        {
          name && <div className="name">{name}</div>
        }
        {
          content && <div className="message">{content}</div>
        }
      </div>
    </div>
  );
}