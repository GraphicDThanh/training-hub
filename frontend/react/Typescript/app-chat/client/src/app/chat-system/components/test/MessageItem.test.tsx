import * as React from 'react';
import { shallow, render } from 'enzyme';
import { MessageItem } from '../MessageItem';
import { message, currentUser } from '../../../../mock/';

const props = {
  currentUser,
  message
};
describe('TEST COMPONENT MESSAGE ITEM', () => {
  it('exist', () => {
    const renderComponent = render(<MessageItem {...props}/>);
    expect(renderComponent).toMatchSnapshot();
  });

  it('Message Item content', () => {
    const {
      author: { id, avatar, name },
      content
    } = message,
    { id: currentId, avatar: currentAvatar } = currentUser;

    const wrapper = shallow(<MessageItem {...props}/>);
    const messageItemWrapper = wrapper.find('.message-item');

    // render right values: avatar, message, name, owner
    expect(messageItemWrapper.find('.avatar').prop('src')).toEqual(avatar ? avatar : currentAvatar)
    expect(messageItemWrapper.find('.message').text()).toEqual(content);
    expect(messageItemWrapper.find('.name').text()).toEqual(name);
    if (id === currentId) {
      expect(wrapper.exists('.owner')).toEqual(true);
    }
  });
});