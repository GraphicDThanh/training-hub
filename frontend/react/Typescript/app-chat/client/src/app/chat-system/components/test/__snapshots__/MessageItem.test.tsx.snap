// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST COMPONENT MESSAGE ITEM exist 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "message-item owner",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "avatar-wrapper",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "avatar user",
              "class": "avatar",
              "src": "https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "info",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "name",
              },
              "children": Array [
                Object {
                  "data": "Thanh",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "message",
                },
                "children": Array [
                  Object {
                    "data": "Hello",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "message",
              },
              "children": Array [
                Object {
                  "data": "Hello",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "name",
                },
                "children": Array [
                  Object {
                    "data": "Thanh",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "info",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "name",
            },
            "children": Array [
              Object {
                "data": "Thanh",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "message",
              },
              "children": Array [
                Object {
                  "data": "Hello",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "message",
            },
            "children": Array [
              Object {
                "data": "Hello",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "name",
              },
              "children": Array [
                Object {
                  "data": "Thanh",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "avatar-wrapper",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "alt": "avatar user",
                "class": "avatar",
                "src": "https://i.pinimg.com/236x/06/a1/a7/06a1a73285a1c3451d9ecc632e1be386--avatar.jpg",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "alt": undefined,
                "class": undefined,
                "src": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
